sdcm/cluster_aws.py

```diff
@@ -333,7 +336,14 @@ def restart(self):
         self.log.debug('Got new public IP %s',
                        self._instance.public_ip_address)
         self.remoter.hostname = self._instance.public_ip_address
-        self.wait_db_up()
+
+    def reboot(self, hard=True):
+        if hard:
+            self.log.debug('Hardly rebooting node')
+            self._instance.reboot()
```

Comment:
reboot is an async operation, it doesn't block until the instance is rebooted [Details](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html#EC2.ServiceResource.reboot)
That means that instance will highly likely not reboot when the reboot method returns. 
Some implementation needed to query instance state

Reply:
good catch, I did it only for gce and missed AWS.

Reply:
`waitsafe` helps here only in case when there is API error and there is a need to retry the API call but doesn't not handle the case that the call is *async*. You need to query instance state and wait untill it reboots or run `uptime` command in the instance

Reply:
it's tricky, instance state isn't changed and with uptime it's a bit ugly.
I'll check what is the response we should get.